File: src/test/java/howtoprogram/words/test/WordCheckerTest.java
package howtoprogram.words.test;

import howtoprogram.words.Dictionary;
import howtoprogram.words.WordChecker;
import org.junit.Assert;
import org.junit.Test;

import java.io.IOException;
import java.nio.file.Path;
import java.nio.file.Paths;

public class WordCheckerTest {
    @Test
    public void invalidWordTest() throws IOException {
        Path path = Paths.get("src/test/resources/books/FiveWords.txt");
        Dictionary dictionary = new Dictionary(path);
        WordChecker checker = new WordChecker(dictionary);
        Assert.assertFalse(checker.isPreviouslyUnseenValidWord("aardwolf"));
    }

    @Test
    public void validWordTest() throws IOException {
        Path path = Paths.get("src/test/resources/books/FiveWords.txt");
        Dictionary dictionary = new Dictionary(path);
        WordChecker checker = new WordChecker(dictionary);
        Assert.assertTrue(checker.isPreviouslyUnseenValidWord("bat"));
        Assert.assertFalse(checker.isPreviouslyUnseenValidWord("bat"));
    }
}

File: src/test/kotlin/pfb/words/test/WordCheckerTest.kt
package pfb.words.test

import org.junit.Assert
import org.junit.Test
import pfb.words.Dictionary
import java.nio.file.Paths
import pfb.words.WordChecker

class WordCheckerTest {
    @Test
    fun notInDictionaryTest() {
        val path = Paths.get("src/test/resources/books/FiveWords.txt")
        val dictionary = Dictionary(path)
        val checker = WordChecker(dictionary)
        Assert.assertFalse(checker.isPreviouslyUnseenValidWord("aardworlff"))
    }

    @Test
    fun validWordTest() {
        val path = Paths.get("src/test/resources/books/FiveWords.txt")
        val dictionary = Dictionary(path)
        val checker = WordChecker(dictionary)
        Assert.assertTrue(checker.isPreviouslyUnseenValidWord("bat"))
        Assert.assertFalse(checker.isPreviouslyUnseenValidWord("bat"))
    }
}